internal/storage/download.go

```diff
@@ -143,6 +139,47 @@ func resume(to string, offset int64, artifact *Artifact, progress progressBytes,
 	return validate(to, artifact.HashType, artifact.HashValue)
 }
 
+func requestDownload(link string, offset int64, serverCert string) (*http.Response, error) {
+	// Create new HTTP request with Range header.
+	request, err := http.NewRequest(http.MethodGet, link, nil)
+	if err != nil {
+		logger.Errorf("Error doing http(s) request to %s", link)
+		return nil, err
+	}
+	if offset > 0 {
+		request.Header.Set("Range", fmt.Sprintf("bytes=%v-", offset))
+	}
+
+	var transport http.Transport
+	var caCertPool *x509.CertPool
+	if len(serverCert) > 0 {
+		caCert, err := ioutil.ReadFile(serverCert)
+		if err != nil {
+			logger.Errorf("Error reading CA certificate file - \"%s\"", serverCert)
+			return nil, err
+		}
+		caCertPool = x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
+	}
+
+	u, _ := url.Parse(link) // MUST not return error, since http(s) request was done to that url
+	if u.Scheme == "https" {
+		config := &tls.Config{ // using the system CA pool
```

Comment:
"using the system CA pool" - is this comment valid? RootCAs are set and the system CA pool should not be used or I am missing something?